app/sdil/controllers/ReturnsController.scala

```diff
@@ -112,56 +115,79 @@ class ReturnsController (
+
+    val data = d(sdilReturn)
+    val subtotal = calculateSubtotal(data)
+    val total: BigDecimal = subtotal + broughtForward
 
     val inner = uniform.fragments.returnsCYA(
-      key = key,
-      lineItems = data,
+      key,
+      data,
       costLower,
       costHigher,
-      subtotal = subtotal,
-      broughtForward = broughtForward,
-      total = subtotal + broughtForward)
+      subtotal,
+      broughtForward,
+      total)
     tell(key, inner)
   }
 
-  private def askReturn(implicit hc: HeaderCarrier): WebMonad[SdilReturn] = (
-    askEmptyOption(litreagePair.nonEmpty, "own-brands-packaged-at-own-sites"),
-    askEmptyOption(litreagePair.nonEmpty, "packaged-as-a-contract-packer"),
-    manyT("small-producer-details", {ask(smallProducer, _)}, min = 1) emptyUnless ask(bool, "exemptions-for-small-producers"),
-    askEmptyOption(litreagePair.nonEmpty, "brought-into-uk"),
-    askEmptyOption(litreagePair.nonEmpty, "brought-into-uk-from-small-producers"),
-    askEmptyOption(litreagePair.nonEmpty, "claim-credits-for-exports"),
-    askEmptyOption(litreagePair.nonEmpty, "claim-credits-for-lost-damaged")
-  ).mapN(SdilReturn.apply)
-
-  private def confirmationPage(key: String, period: ReturnPeriod, subscription: Subscription)(implicit messages: Messages): WebMonad[Result] = {
+  private def confirmationPage(key: String, period: ReturnPeriod, subscription: Subscription,
+                               sdilReturn: SdilReturn, broughtForward: BigDecimal, sdilRef: String)(implicit messages: Messages): WebMonad[Result] = {
     val now = LocalDate.now
     import ltbs.play._
+
+    val data = d(sdilReturn)
+    val subtotal = calculateSubtotal(data)
+
+    val total = subtotal + broughtForward
+
+    def formatMoney (total: BigDecimal) = {
+      if(total < 0)
+        f"-£${total.abs}%,.2f"
+       else
+        f"£$total%,.2f"
+      }
+
+    val getTotal = messages("return-sent.subheading", formatMoney(total), period.deadline.format("dd MMMM yyyy"))
+
     val returnDate = messages(
       "return-sent.returnsDoneMessage",
       period.start.format("MMMM"),
       period.end.format("MMMM"),
       period.start.getYear.toString,
       subscription.orgName,
-      LocalTime.now.format(DateTimeFormatter.ofPattern("HH:mm")),
-      now.format("dd! MMMM")
+      LocalTime.now.format(DateTimeFormatter.ofPattern("h:mma")).toLowerCase,
+      now.format("dd MMMM yyyy")
     )
 
-    val whatHappensNext = uniform.fragments.returnsPaymentsBlurb(period.deadline)(messages).some
-    journeyEnd(key, now, Html(returnDate).some, whatHappensNext)
+    val whatHappensNext = uniform.fragments.returnsPaymentsBlurb(period, sdilRef)(messages).some
+    journeyEnd(key, now, Html(returnDate).some, whatHappensNext, Html(getTotal).some)
   }
 
-  private def program(period: ReturnPeriod, subscription: Subscription)(implicit hc: HeaderCarrier): WebMonad[Result] = for {
+  private def askReturn(implicit hc: HeaderCarrier): WebMonad[SdilReturn] = (
```

Comment:
swapping the order of askReturn and program makes it hard for me to see what has changed